lib/entities/game.rb

```diff
@@ -0,0 +1,112 @@
+class Game
+  include Validation
+
+  AMOUNT_DIGITS = 4
+  RANGE_OF_DIGITS = { first: 0, last: 6 }.freeze
+  DIFFICULTY_LEVELS = {
+                        easy: 'Easy',
+                        hard: 'Hard',
+                        expert: 'Expert' }.freeze
+  GUESS_CODE = { hint: 'hint', leave: 'exit' }.freeze
+
+  attr_reader :name, :hints_total, :attempts_total, :hints_used, :attempts_used, :difficulty, :winner, :attempts_left
+  attr_accessor :errors
+  def initialize
+    @errors = []
+    @hints_used = 0
+    @attempts_used = 0
+    @secret_code = (0...AMOUNT_DIGITS).map { rand(RANGE_OF_DIGITS[:first]..RANGE_OF_DIGITS[:last]) }
+    @arr_for_hints = @secret_code.map(&:dup).shuffle(random: Random.new(1))
+  end
+
+  def try(user_inputed_code)
+    return use_hint if user_inputed_code == GUESS_CODE[:hint]
+
+    verdict(user_inputed_code.split('').map(&:to_i)) if check_input(user_inputed_code)
+  end
+
+  def set_up_difficulty(difficulty, name)
+    @name = name
+    case difficulty
```

Comment:
You can return by case object with expected params and assign them to @variables.
Then you  can remove methods easy hard expert.